lib/mp4/transmuxer.js

```diff
@@ -929,20 +939,44 @@ CoalesceStream.prototype.flush = function(flushSource) {
 Transmuxer = function(options) {
   var
     self = this,
+    hasFlushed = true,
     videoTrack,
     audioTrack,
     packetStream, parseStream, elementaryStream,
     adtsStream, h264Stream,aacStream,
     videoSegmentStream, audioSegmentStream, captionStream,
     coalesceStream,
-    headOfPipeline;
+    headOfPipeline,
+    emitData,
+    emitDone;
 
   Transmuxer.prototype.init.call(this);
 
   options = options || {};
   this.baseMediaDecodeTime = options.baseMediaDecodeTime || 0;
+  this.currentPipeline_ = null;
+
+  this.teardownAacPipeline = function() {
```

Comment:
Instead of these teardown functions, what about just tossing out the old pipeline and creating a new one?